app/lib/actions/staypuft/hostgroup/ordered_deploy.rb

```diff
@@ -44,34 +44,56 @@ def plan(hostgroups, hosts_to_deploy_filter, hosts_to_provision_filter)
               input[:hostgroups].update hostgroup.id => { name: hostgroup.name, hosts: {} }
               hostgroup_hosts = (hostgroup.hosts & hosts_to_deploy_filter)
 
-              # wait till all hosts are ready
-              concurrence do
-                hostgroup_hosts.each do |host|
-                  input[:hostgroups][hostgroup.id][:hosts].update host.id => host.name
-
-                  plan_action Host::WaitUntilReady, host
-                  plan_action Host::Update, host, :environment => nil
-                end
+              case hostgroup.role.orchestration
+              when 'leader'
+                configure_hosts_leader(hostgroup, hostgroup_hosts)
+              else
+                configure_hosts_concurrent(hostgroup, hostgroup_hosts)
```

Comment:
I would include a FIXME comment here -- it's not handling the 'serial' type of orchestration. We don't have any current roles that use that, and with suspension of active development, we probably won't nee it, but we should at least comment here that the valid setting 'serial' isn't being honored here right now.